### PR TITLE
Add DisableVMAgent parameter to Set-AzureRmVMOperatingSystem cmdlet

### PR DESCRIPTION
**Pull request by hyonholee**
This PR resolves the following issue:
https://github.com/Azure/azure-powershell/issues/5838

https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/42

## Description

Add DisableVMAgent parameter to Set-AzureRmVMOperatingSystem cmdlets, and fix the behavior when neither ProvisionVMAgent nor DisableVMAgent are given.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by markcowl** [comment]
@hyonholee Provision / Disable VM Agent should be mutually exclusive by parameter set

**Comment by hyonholee** [comment]
@markcowl I updated the PR by adding two additional parameter sets so that Provision and Disable Agent are mutually exclusive.  Thanks.

File: src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Config/SetAzureVMOperatingSystemCommand.cs [hunk 1]
```diff
@@ -238,7 +303,18 @@ public override void ExecuteCmdlet()
...
+                {
+                    this.VM.OSProfile.WindowsConfiguration.ProvisionVMAgent = true;
+                }
+                else if (this.DisableVMAgent.IsPresent)
+                {
+                    this.VM.OSProfile.WindowsConfiguration.ProvisionVMAgent = false;
+                }
+                else
+                {
+                    this.VM.OSProfile.WindowsConfiguration.ProvisionVMAgent = this.VM.OSProfile.WindowsConfiguration.ProvisionVMAgent;
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Config/SetAzureVMOperatingSystemCommand.cs]: @hyonholee is this `else` block necessary? It looks like we just need the `if` and `else-if` blocks

**Comment by hyonholee** [comment]
@markcowl  could you merge if there is no more concern?  Thanks.

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/750/

**Pull request merged**